pkg/square/builder.go

```diff
@@ -410,10 +409,15 @@ func (e element) maxShareOffset() int {
 	return e.numShares + e.maxPadding
 }
 
-func worstCaseShareIndexes(blobs, maxSquareCapacity int) []uint32 {
+// worstCaseShareIndexes returns the largest possible shared indexes for a set
+// of blobs at a given appversion. Largest possible is "worst" in that protobuf
+// uses varints to encode integers, so larger integers can require more bytes to
+// encode.
+func worstCaseShareIndexes(blobs int, appVersion uint64) []uint32 {
+	maxSquareSize := appconsts.SquareSizeUpperBound(appVersion)
 	shareIndexes := make([]uint32, blobs)
 	for i := range shareIndexes {
-		shareIndexes[i] = uint32(maxSquareCapacity)
+		shareIndexes[i] = uint32(maxSquareSize * maxSquareSize)
 	}
 	return shareIndexes
 }
```

Comment:
Pls review!!

here's the bug, where in some edge cases (lots of txs) we'd end up using different values for the worst case, which would cause different square constructions when using different "maxSquareSize" values